File: problem_seven/7.rb
def do_magic(term)
  [].tap do |primes|
    num = 2
    until primes.length >= term
      primes << num if is_prime?(num)
      num += 1
    end
  end.last
end

def is_prime?(num)
  [*2...num].all? { |divisor| num % divisor != 0 }
end

puts do_magic(10001)

File: problem_seven/7.js
(function doMagic (term) {
  primes = storePrimes([], term);
  result = primes.pop();
  console.log(result);
  return result;
})(10001);

function storePrimes (primes, term) {
  var num = 2;
  while(primes.length <= term){
    if (isPrime(num)) {
      primes.push(num);
    }
    num ++;
  }
  return primes;
}

function isPrime (number) {
  for (var i=2; i<number; i++){
    if (number % number === 0) {
      return false;
    }
  }
  return true;
}